generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FieldCollectionHelper.java

```diff
@@ -0,0 +1,169 @@
+        //is the field for this atomic constraint fixed?
+        //does the constraint complement or conflict with the fixed field?
+
+        Field field = atomicConstraint.getField();
+        FixedField fixedFieldValue = fieldCollection.getFixedField(field);
+        if (fixedFieldValue == null){
+            //field isn't fixed
+            return AtomicConstraintFixedFieldBehaviour.FIELD_NOT_FIXED;
+        }
+
+        //field is fixed, work out if it is contradictory
+        return fixedValueConflictsWithAtomicConstraint(fixedFieldValue, atomicConstraint)
+            ? AtomicConstraintFixedFieldBehaviour.CONSTRAINT_CONTRADICTS
+            : AtomicConstraintFixedFieldBehaviour.NON_CONTRADICTORY;
+    }
+
+    //create a mapping of field->fieldspec for each fixed field - efficiency
+    private Map<Field, FieldSpec> getFieldSpecsForAllFixedFieldsExceptLast(FieldCollection fieldCollection, ConstraintNode constraintNode){
+        Map<Field, List<AtomicConstraint>> fieldToConstraints = constraintNode.getAtomicConstraints()
+            .stream()
+            .collect(Collectors.groupingBy(AtomicConstraint::getField));
+
+        return fieldCollection.getFixedFields().values()
+            .stream()
+            .collect(Collectors.toMap(
+                ff -> ff.field,
+                ff -> {
+                    FieldSpec fieldSpec = getFieldSpec(ff, fieldToConstraints.get(ff.field));
+                    return fieldSpec == null
+                        ? FieldSpec.Empty
+                        : fieldSpec;
+                }
+            ));
+    }
+
+
+    //TODO move this, as it is all fixedField related, and not field collection stuff
+    //for the given field get a stream of possible values
+    private FixedField getFixedFieldWithValuesForField(Field field, ConstraintNode rootNode) {
+        //from the original tree, get all atomic constraints that match the given field
+        Set<AtomicConstraint> constraintsForRootNode = rootNode.getAtomicConstraints()
+            .stream()
+            .filter(c -> c.getField().equals(field))
+            .collect(Collectors.toSet());
+
+        //produce a fieldspec for all the atomic constraints
+        FieldSpec rootConstraintsFieldSpec = this.reducer.reduceConstraintsToFieldSpec(constraintsForRootNode)
+            .orElse(FieldSpec.Empty);
+
+        //use the FieldSpecFulfiller to emit all possible values given the generation mode, interesting or full-sequential
+        Stream<Object> values = new FieldSpecFulfiller(field, rootConstraintsFieldSpec)
+            .generate(this.generationConfig)
+            .map(dataBag -> dataBag.getValue(field));
+
+        return new FixedField(field, values, rootConstraintsFieldSpec, this.monitor);
+    }
+    //work out if the field is contradictory
+    private boolean fixedValueConflictsWithAtomicConstraint(FixedField fixedField, AtomicConstraint atomicConstraint) {
```

Comment:
This method could move to the `ReductiveDecisionTreeAdapter` since the previously private methods are now public, same as `shouldIncludeAtomicConstraint`